lib/syntax.ml

```diff
@@ -124,9 +124,15 @@ let exists_grouping columns =
 
 (* all columns from tables, without duplicates *)
 (* FIXME check type of duplicates *)
-let make_unique = List.unique ~cmp:Schema.Source.Attr.(fun a1 a2 -> a1.attr.name = a2.attr.name && a1.attr.name <> "")
+let make_unique =
+  List.unique ~cmp:(fun a1 a2 ->
+    (* Check if columns are from the same table (source)  *)
+    a1.Schema.Source.Attr.sources = a2.sources
+    && a1.attr.name = a2.attr.name
+    (* Check if columns are unnamed (for example a function is applied to a column) and unaliased *)
```

Comment:
this comment sounds wrong, because it actually checks that columns _are named_

Reply:
```SELECT MAX(col) FROM smth```
in this case, it checks `if name <> ""`, because `if name = ""` we skip it (consider them as unnamed) because we can have more than one field such that, so yes, you are right.